platform/android/SCsub

```diff
@@ -55,3 +55,21 @@ if lib_arch_dir != '':
 
     stl_lib_path = str(env['ANDROID_NDK_ROOT']) + '/sources/cxx-stl/llvm-libc++/libs/' + lib_arch_dir + '/libc++_shared.so'
     env_android.Command(out_dir + '/libc++_shared.so', stl_lib_path, Copy("$TARGET", "$SOURCE"))
+
+# Zip android/java folder for the source export template.
+print("Archiving platform/android/java as bin/android_source.zip...")
+import os
+import zipfile
+# Change dir to avoid have zipped paths start from the android/java folder.
+olddir = os.getcwd()
+os.chdir(Dir('#platform/android/java').abspath)
+zipf = zipfile.ZipFile(File('#bin/android_source.zip').abspath, 'w', zipfile.ZIP_DEFLATED)
```

Comment:
Looks like this runs early and fails if the `bin` folder wasn't created yet.
I'll wait to see if we go with another approach before reworking it.